### ISSUE TITLE
Update GoCloak interface in README

### ISSUE THREAD

**Issue by Zamony** [Nerzal/gocloak#223]
The documentation is misleading. For example:
`CreateGroup(ctx context.Context, accessToken, realm string, group Group) error`
The correct definition in gocloak.go is
`CreateGroup(ctx context.Context, accessToken, realm string, group Group) (string, error)`

**Comment by timdrysdale** [comment]
I've created a [small tool](https://github.com/timdrysdale/ifcmp) to compare the README and a file containing the interface, via their [AST](https://golang.org/pkg/go/ast/). This indicates the current inconsistencies are: 
```
Actual: GetResource(ctx context.Context, token, realm, clientID, resourceID string) (*ResourceRepresentation, error)
Readme: GetResource(ctx context.Context, token, realm, clientID, resourceID string) (*Resource, error)

Actual: DecodeAccessTokenCustomClaims(ctx context.Context, accessToken, realm, expectedAudience string, claims jwt.Claims) (*jwt.Token, error)
Readme: DecodeAccessTokenCustomClaims(ctx context.Context, accessToken, realm string, claims jwt.Claims) (*jwt.Token, error)

Actual: CreateClientScope(ctx context.Context, accessToken, realm string, scope ClientScope) (string, error)
Readme: CreateClientScope(ctx context.Context, accessToken, realm string, scope ClientScope) error

Actual: CreateRealmRole(ctx context.Context, token, realm string, role Role) (string, error)
Readme: CreateRealmRole(ctx context.Context, token, realm string, role Role) error

Actual: DeletePolicy(ctx context.Context, token, realm, clientID, policyID string) error
Readme: DeletePolicy(ctx context.Context, token, realm, clientID string, policyID string) error

Actual: CreateClientRole(ctx context.Context, accessToken, realm, clientID string, role Role) (string, error)
Readme: CreateClientRole(ctx context.Context, accessToken, realm, clientID string, role Role) error

Actual: UpdateResource(ctx context.Context, token, realm, clientID string, resource ResourceRepresentation) error
Readme: UpdateResource(ctx context.Context, token, realm, clientID string, resource Resource) error

Actual: CreateResource(ctx context.Context, token, realm, clientID string, resource ResourceRepresentation) (*ResourceRepresentation, error)
Readme: CreateResource(ctx context.Context, token, realm, clientID string, resource Resource) (*Resource, error)

Actual: CreateClient(ctx context.Context, accessToken, realm string, clientID Client) (string, error)
Readme: CreateClient(ctx context.Context, accessToken, realm string, clientID Client) error

Actual: CreateClientProtocolMapper(ctx context.Context, token, realm, clientID string, mapper ProtocolMapperRepresentation) (string, error)
Readme: CreateClientProtocolMapper(ctx context.Context, token, realm, clientID string, mapper ProtocolMapperRepresentation) error

Actual: LoginClientSignedJWT(ctx context.Context, clientID, realm string, key interface{}, signedMethod jwt.SigningMethod, expiresAt *jwt.Time) (*JWT, error)
Readme: LoginClientSignedJWT(ctx context.Context, clientID, realm string, key interface{}, signedMethod jwt.SigningMethod, expiresAt int64) (*JWT, error)

Actual: DecodeAccessToken(ctx context.Context, accessToken, realm, expectedAudience string) (*jwt.Token, *jwt.MapClaims, error)
Readme: DecodeAccessToken(ctx context.Context, accessToken, realm string) (*jwt.Token, *jwt.MapClaims, error)

Actual: CreateGroup(ctx context.Context, accessToken, realm string, group Group) (string, error)
Readme: CreateGroup(ctx context.Context, accessToken, realm string, group Group) error

Actual: CreateComponent(ctx context.Context, accessToken, realm string, component Component) (string, error)
Readme: CreateComponent(ctx context.Context, accessToken string, realm, component Component) error

Actual: GetResources(ctx context.Context, token, realm, clientID string, params GetResourceParams) ([]*ResourceRepresentation, error)
Readme: GetResources(ctx context.Context, token, realm, clientID string) ([]*Resource, error)
```
Later on, I'll make a PR with the updated README fixing these.

**Comment by timdrysdale** [comment]
done: [PR #240](https://github.com/Nerzal/gocloak/pull/240)

**Comment by GabriFila** [comment]
Hi everyone!

I believe there is another method that is missing from the README:

[Line 215](https://github.com/Nerzal/gocloak/blob/4ce82f58b38c36ca0f8704c8f6384a975b1fd6f6/gocloak.go#L215)
```go
// AddClientRoleToUser adds a client role to the user
AddClientRoleToUser(ctx context.Context, token, realm, clientID, userID string, roles []Role) error
```

Hope this was helpful!

**Comment by Nerzal** [comment]
Btw - this tool is awesome @timdrysdale :)

**Comment by timdrysdale** [comment]
@GabriFila good spot - I forgot to report missing functions. I've updated the tool, and there are a few...
```
Actual: UpdateIdentityProvider(ctx context.Context, token, realm, alias string, providerRep IdentityProviderRepresentation) error
Readme: 

Actual: GetPermissionResources(ctx context.Context, token, realm, clientID, permissionID string) ([]*PermissionResource, error)
Readme: 

Actual: GetPermissionScopes(ctx context.Context, token, realm, clientID, permissionID string) ([]*PermissionScope, error)
Readme: 

Actual: GetIdentityProvider(ctx context.Context, token, realm, alias string) (*IdentityProviderRepresentation, error)
Readme: 

Actual: GetClientRoleByID(ctx context.Context, accessToken, realm, roleID string) (*Role, error)
Readme: 

Actual: DeleteIdentityProvider(ctx context.Context, token, realm, alias string) error
Readme: 

Actual: CreateIdentityProvider(ctx context.Context, token, realm string, providerRep IdentityProviderRepresentation) (string, error)
Readme: 

Actual: GetToken(ctx context.Context, realm string, options TokenOptions) (*JWT, error)
Readme: 

Actual: GetUserFederatedIdentities(ctx context.Context, token, realm, userID string) ([]*FederatedIdentityRepresentation, error)
Readme: 

Actual: AddClientRoleToUser(ctx context.Context, token, realm, clientID, userID string, roles []Role) error
Readme: 

Actual: DeleteUserFederatedIdentity(ctx context.Context, token, realm, userID, providerID string) error
Readme: 

Actual: DeleteUserFromGroup(ctx context.Context, token, realm, userID, groupID string) error
Readme: 

Actual: CreateUserFederatedIdentity(ctx context.Context, token, realm, userID, providerID string, federatedIdentityRep FederatedIdentityRepresentation) error
Readme: 

Actual: GetIdentityProviders(ctx context.Context, token, realm string) ([]*IdentityProviderRepresentation, error)
Readme: 

Actual: RestyClient() *resty.Client
Readme: 

Actual: GetDependentPermissions(ctx context.Context, token, realm, clientID, policyID string) ([]*PermissionRepresentation, error)
Readme: 

Actual: DeleteClientRoleFromUser(ctx context.Context, token, realm, clientID, userID string, roles []Role) error
Readme: 

Actual: SetRestyClient(restyClient *resty.Client)
Readme: 

```
I'll get these added to the README and submit another PR ....

**Comment by timdrysdale** [comment]
Now fixed in [PR #241](https://github.com/Nerzal/gocloak/pull/241) - should be able to close this issue. @Nerzal let me know if you want any changes to [ifcmp](https://github.com/timdrysdale/ifcmp) to enable including it in CI.
